sympy/solvers/ode/riccati.py

```diff
@@ -0,0 +1,857 @@
+
+    2. We can then assume that the Laurent series of `g(x)`
+    takes the following form -
+
+    .. math:: g(x) = \frac{num(x)}{den(x)} = \Sum_{m = 0}^{\infty} a_m x^m
+
+    where `a_m` denotes the Laurent series coefficients.
+
+    3. Multiply the denominator to the RHS of the equation
+    and form a recurrence relation for the coefficients `a_m`.
+    """
+    one = Poly(1, x, extension=True)
+
+    if r == oo:
+        # Series at x = oo is equal to first transforming
+        # the function from x -> 1/x and finding the
+        # series at x = 0
+        num, den = inverse_transform_poly(num, den, x)
+        r = S(0)
+
+    if r:
+        # For an expansion about a non-zero point, a
+        # transformation from x -> x + r must be made
+        num = num.transform(Poly(x + r, x, extension=True), one)
+        den = den.transform(Poly(x + r, x, extension=True), one)
+
+    # Remove the pole from the denominator if the series
+    # expansion is about one of the poles
+    num, den = (num*x**m).cancel(den, include=True)
+
+    # Equate coefficients for the first terms (base case)
+    maxdegree = 1 + max(num.degree(), den.degree())
+    syms = symbols(f'a:{maxdegree}', cls=Dummy)
+    diff = num - den * Poly(syms[::-1], x)
+    coeff_diffs = diff.all_coeffs()[::-1][:maxdegree]
+    (coeffs, ) = linsolve(coeff_diffs, syms)
+
+    # Use the recursion relation for the rest
+    recursion = den.all_coeffs()[::-1]
+    div, rec_rhs = recursion[0], recursion[1:]
+    series = list(coeffs)
+    while len(series) < n:
+        next_coeff = sum(c*series[-1-n] for n, c in enumerate(rec_rhs)) / div
+        series.append(-next_coeff)
+    series = {m - i: val for i, val in enumerate(series)}
+    return series
+
+def compute_m_ybar(x, poles, choice, c, d, N):
+    """
+    Helper function to calculate -
+
+    1. m - The degree bound for the polynomial
+    solution that must be found for the auxiliary
+    differential equation.
+
+    2. ybar - Part of the solution which can be
+    computed using the poles, c and d vectors.
+    """
+    ybar = 0
+    m = Poly(d[choice[0]][-1], x, extension=True)
+
+    # Calculate the first (nested) summation for ybar
+    # as given in Step 9 of the Thesis (Pg 82)
+    for i in range(len(poles)):
+        for j in range(len(c[i][choice[i + 1]])):
+            ybar += c[i][choice[i + 1]][j]/(x - poles[i])**(j+1)
+        m -= Poly(c[i][choice[i + 1]][0], x, extension=True)
+
+    # Calculate the second summation for ybar
+    for i in range(N+1):
+        ybar += d[choice[0]][i]*x**i
+    return (m.expr, ybar)
+
+
+def solve_aux_eq(numa, dena, numy, deny, x, m):
+    """
+    Helper function to find a polynomial solution
+    of degree m for the auxiliary differential
+    equation.
+    """
+    # Assume that the solution is of the type
+    # p(x) = C_0 + C_1*x + ... + C_{m-1}*x**(m-1) + x**m
+    psyms = symbols(f'C0:{m}', cls=Dummy)
+    psol = Poly((psyms + (1,))[::-1], x, domain=ZZ[psyms])
```

Comment:
This can be quite slow for a large number of generators:
```python
In [63]: psyms = symbols('C:1000')

In [64]: K = ZZ[psyms]

In [65]: %time p1 = Poly(psyms, t, domain=K)
CPU times: user 530 ms, sys: 27.7 ms, total: 557 ms
Wall time: 562 ms

In [66]: %time p2 = Poly(K.gens, t, domain=K)
CPU times: user 39.4 ms, sys: 295 µs, total: 39.7 ms
Wall time: 40.2 ms

In [67]: p1 == p2
Out[67]: True
```